src/uniprotkb/components/entry/GoRibbon.tsx

```diff
@@ -75,6 +75,38 @@ const GoRibbon = ({
 
   const [elementLoaded, setElementLoaded] = useSafeState(false);
 
+  useEffect(() => {
+    if (organismData?.scientificName && organismData?.lineage) {
+      const taxonomyInfo = [
+        ...organismData.lineage,
+        organismData.scientificName,
+      ];
+
+      // SlimSets based on Taxonomy
+      const slimSetByTaxon = {
+        goslim_plant: [
+          'Viridiplantae',
+          'Bangiophyceae',
+          'Florideophyceae',
+          'Stylonematophyceae',
+          'Rhodellophyceae',
+          'Compsopogonophyceae',
+        ],
+        prokaryotes: ['Bacteria', 'Archaea'],
+      };
+
+      // Check if the taxon matches a slimset
+      Object.entries(slimSetByTaxon).forEach(([key, value]) => {
+        const presentTaxon = taxonomyInfo?.filter(
+          (t) => value.includes(String(t)) // Lineage is Array of strings here
+        );
+        if (presentTaxon?.length) {
+          setSelectedSet(key);
+        }
+      });
+    }
+  }, [organismData]);
```

Comment:
This function is not really an effect, it's more of an initial state, so I would pass this function as a state initialiser to the useState on line 68, returning the found key, or if none is found `'goslim_generic'`

Reply:
:woman-facepalming: 
I did the change